### ISSUE TITLE
MakeGood / Eclipse support

### ISSUE THREAD

**Issue by WolfgangFahl** [SemanticMediaWiki/SemanticMediaWiki#1333]
To use Eclipse and XDebug as a development environment there is an ecplise plugin-in "MakeGood" that helps with running and debugging phpunit tests.

I had some success in getting this to work see:
see https://www.semantic-mediawiki.org/wiki/User:WolfgangFahl/Workdocumentation_2015-12-27

It would be great if we could get this to a state where more people can use it.

**Comment by WolfgangFahl** [comment]
https://www.semantic-mediawiki.org/wiki/SMW_Development_with_Eclipse now has a description for the context in which this would be useful. Tentatively I could create a pull request containing MakeGoodPreload.php and globals.inc. Any opions? 

I'd be glad to get rid of globals.inc - Any idea how this could be done?

**Comment by mwjames** [comment]
> I'd be glad to get rid of globals.inc - Any idea how this could be done?

No idea. I use sublime and it works out-of the box without any special arrangements. @JeroenDeDauw is using PHPStorm and as far as I know hadn't had any issues.

> Tentatively I could create a pull request containing MakeGoodPreload.php and globals.inc. Any opinons? 

The code base is tool agnostic (except for Travis but that is a general exception) and seems [0, 1] didn't require special handling or?

[0] http://biowikifarm.net/meta/Installation_of_PHP_%26_Eclipse_to_debug_Mediawiki_(Windows)
[1] http://saml.rilspace.org/debugging-mediawiki-extensions-with-eclipse-for-php-and-xdebug-on-ubuntu-1004

**Comment by WolfgangFahl** [comment]
Thanks for looking into this MakeGood is for PHPUnit Integration. The debugging works almost out of the box.

**Comment by JeroenDeDauw** [comment]
We can always include a helper file for a specific tool, as long as it does not bug the rest of the codebase, it is a reasonable maintenance cost, and has someone that maintains it (as opposed to just adding it).

> @JeroenDeDauw is using PHPStorm and as far as I know hadn't had any issues.

I used PhpSTorms PHPUnit integration two years back or so. However due to problems the IDE not adopting new versions of PHPUnit quick enough I've since switched back to just using the CLI. That is, for most of the things I work on. For SMW I do not have a working local setup so am not running tests locally at all.

**Comment by WolfgangFahl** [comment]
Yes - I am willing to support the MakeGood Preloader - in principle it should only be a few lines of code. Should this go e.g. in a "contrib" subfolder to clarify that it's not part of the main code base?

The dockerized SMW is a good starting point for having a local setup available quickly. It's even possible to run that in a docker container e.g. from digitalcloud for just a few cents per hour. Unfortunately the performance is not very good but for a specific test or problem it might be in many cases good enough.

**Comment by mwjames** [comment]
> We can always include a helper file for a specific tool, as long as it does not bug the rest of the codebase
> Should this go e.g. in a "contrib" subfolder to clarify that it's not part of the main code base?

If @JeroenDeDauw thinks this can be included, I don't mind as long as it doesn't appear in the root folder (where `src` or `tests` is placed).

If it test related then it should go under `tests` anything else should find its place in either `docs` or `build`.
